x/concentrated-liquidity/incentives.go

```diff
@@ -709,6 +709,34 @@ func prepareAccumAndClaimRewards(accum accum.AccumulatorObject, positionKey stri
 	return incentivesClaimedCurrAccum, dust, nil
 }
 
+// moveRewardsToNewPositionAndDeleteOld claims the rewards from the old position and moves them to the new position.
+// This has a side-effect of deleting the old position (happens osmoutils/accum.ClaimRewards when number of shares is zero)
+// The positions must be associated with the given accumulator.
+// The given growth outside the positions range is used for claim rewards accounting.
+// The rewards are moved as "unclaimed rewards" to the new position.
+// Returns nil on success. Error otherwise.
+func moveRewardsToNewPositionAndDeleteOld(ctx sdk.Context, accum accum.AccumulatorObject, oldPositionName, newPositionName string, growthOutside sdk.DecCoins) error {
+	if oldPositionName == newPositionName {
+		return types.ModifySamePositionAccumulatorError{PositionAccName: oldPositionName}
+	}
+
+	if err := preparePositionAccumulator(accum, oldPositionName, growthOutside); err != nil {
```

Comment:
No, this seems correct because we want to prepare and claim from the old position accumulator.

Once claimed, the rewards are moved into the "unclaimed rewards field" of the new accumulator.

Thanks for checking this though